### ISSUE TITLE
[Docker] Error 500 when creating Nest on fresh panel install

### ISSUE THREAD

**Issue by Telokis** [pterodactyl/panel#5187]
# Current Behavior

On a fresh install using a `docker-compose.yml` (provided below) with the minimal possible configuration, I am not able to create a Nest. I am met with an error 500 and in the logs it complains about the database:

```
[2024-08-11 16:41:40] production.NOTICE: A query exception was encountered while trying to load settings from the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'panel.settings' doesn't exist (Connection: mysql, SQL: select * from `settings`)
```

However, everything else is fine, on the real deployment I've even been able to deploy Wings and create a game server and connect to it. (But even after connecting to the game server, I still get the 500 when I try to create a Nest).

I can also properly access `/admin/settings/advanced` and see the settings.

I went even further and used the `mysql` cli within the MariaDB container to query the `settings` table within the `panel` table and confirmed it to be present and properly filled.  

The container logs show the migrations being run and succeeding.

(I already asked for help on Discord but no response and this is perfectly reproducible: https://discord.com/channels/122900397965705216/493443683685892096/1272188747953344562)

# Expected Behavior

Well I expected to be able to create a Nest.

# Steps to Reproduce

### Step 1
Use the docker-compose specified below

<details>

<summary>Full docker-compose.yml</summary>

```yml
services:
  mariadb:
    image: mariadb:10.5
    restart: unless-stopped
    command: --default-authentication-plugin=mysql_native_password
    healthcheck:
      test:
        ["CMD-SHELL", "healthcheck.sh --connect --innodb_initialized || exit 1"]
      start_period: 10s
      interval: 10s
      timeout: 1s
      retries: 3
    environment:
      - SERVICE_PASSWORD_MYSQL
      - MYSQL_ROOT_PASSWORD=my_root_db_password
      - MYSQL_DATABASE=panel
      - MYSQL_USER=pterodactyl
      - MYSQL_PASSWORD=my_db_password
    volumes:
      - pterodactyl-db:/var/lib/mysql

  redis:
    image: redis:alpine
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "redis-cli ping || exit 1"]
      interval: 10s
      timeout: 1s
      retries: 3

  pterodactyl:
    image: ghcr.io/pterodactyl/panel:latest
    restart: unless-stopped
    ports:
      - 8738:80
    volumes:
      - "panel-var:/app/var/"
      - "panel-nginx:/etc/nginx/http.d/"
      - "panel-certs:/etc/letsencrypt/"
      - "panel-logs:/app/storage/logs"
    healthcheck:
      test: ["CMD-SHELL", "curl -sf http://localhost:80 || exit 1"]
      interval: 10s
      timeout: 1s
      retries: 3
    environment:
      - APP_ENV=production
      - APP_ENVIRONMENT_ONLY=false
      - APP_URL=http://localhost:8738
      - APP_TIMEZONE=${TIMEZONE:-UTC}
      - APP_SERVICE_AUTHOR=$APP_SERVICE_AUTHOR
      - LOG_LEVEL=${LOG_LEVEL:-debug}

      - CACHE_DRIVER=redis
      - SESSION_DRIVER=redis
      - QUEUE_DRIVER=redis
      - REDIS_HOST=redis

      - DB_HOST=mariadb
      - DB_PORT=3306
      - DB_PASSWORD=my_db_password

      - MAIL_FROM=$MAIL_FROM
      - MAIL_DRIVER=$MAIL_DRIVER
      - MAIL_HOST=$MAIL_HOST
      - MAIL_PORT=$MAIL_PORT
      - MAIL_USERNAME=$MAIL_USERNAME
      - MAIL_PASSWORD=$MAIL_PASSWORD
      - MAIL_ENCRYPTION=$MAIL_ENCRYPTION

volumes:
  panel-var:
  panel-nginx:
  panel-certs:
  panel-logs:
  pterodactyl-db:
```

</details>

### Step 2
Use the following php artisan command to create an admin user (`myadmin`/`mypassword`):

```bash
docker compose exec pterodactyl php artisan p:user:make --no-interaction --admin=1 --email=test@example.com --username=myadmin --name-first=adm --name-last=in --password=mypassword
```

### Step 3
Head to http://localhost:8738 and login with `myadmin`/`mypassword`.  
Then go to http://localhost:8738/admin/nests/new and try to create a Nest.

![image](https://github.com/user-attachments/assets/791c399b-6473-4578-b6fb-5fa652913361)

### Step 4
Go back to the terminal and use the following command to see the logs:
```bash
docker compose exec pterodactyl sh -c "cat /app/storage/logs/*"
```

You will see something like the following:

```
[2024-08-11 16:41:40] production.NOTICE: A query exception was encountered while trying to load settings from the database: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'panel.settings' doesn't exist (Connection: mysql, SQL: select * from `settings`)
```

# Panel Version

[v1.11.7](https://github.com/pterodactyl/panel/pkgs/container/panel/213438609?tag=v1.11.7)

# Wings Version

Not used yet

# Games and/or Eggs Affected

_No response_

# Docker Image

_No response_

# Error Logs

_No response_

# Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by danny6167** [comment]
Hi.
The log entries you're seeing are old and not relevant to any issue you're having.
Good chance the permissions are wrong on your logs folder and it is preventing the logging of the actual issue.
Please correct the permissions, reproduce the issue, and then see if anything new is logged.

``docker compose exec panel chown -R nginx: /app/storage/logs/``

**Comment by Telokis** [comment]
Hi @danny6167 and thanks for the very quick response.

You were right about the logs having the wrong permissions, I'll write that down to figure out later.

Now I have the real error that is quite weird. ([Full log](https://ptero.co/ezojecoyur.shell))
```
[2024-08-11 17:05:15] production.ERROR: Could not save Pterodactyl\Models\Nest[]: failed to validate data: {"author":["The author field is required."]} {"userId":1,"exception":"[object] (Pterodactyl\\Exceptions\\Model\\DataValidationException(code: 0): Could not save Pterodactyl\\Models\\Nest[]: failed to validate data: {\"author\":[\"The author field is required.\"]} at /app/app/Models/Model.php:53)
```

The "Author" field is required but it's not shown on the page?

![image](https://github.com/user-attachments/assets/f146ae88-c5d8-43f3-9c37-5a8d98fb6f03)

**Comment by danny6167** [comment]
Your error suggests that your APP_SERVICE_AUTHOR environment variable in your compose file is either missing or invalid. That's the author it's referring to.

**Comment by Telokis** [comment]
It is indeed unset at the moment, I didn't expect it to be that important.  
And after a few quick tries, it must be set to an email address.

Thank you for the assistance, I would never have found that out.

About the log folder permissions, do you have a link to point me to that could help me fix it by default, please?

**Comment by philhk** [comment]
Hello,

I believe there is more to this issue. Whenever I tried to update anything for my node, I encountered a 500 server error. Here are my [logs](https://ptero.co/ivufefatuh). After running `docker compose exec panel chown -R nginx: /app/storage/logs/`, I was able to update the node without any problems. 

I received a warning: `The daemon configuration has been updated; however, an error was encountered while attempting to automatically update the configuration file on the daemon. You will need to manually update the configuration file (config.yml) for the daemon to apply these changes.` This is likely normal because I haven't set up the daemon yet.

Here are my new [logs](https://ptero.co/efibiwixak). They still indicate a `Connection refused` error for the database, but from the panel's perspective, everything seems to be working fine. I suspect that because the daemon isn't reachable, it throws an exception that it tries to write into the logs but lacks permission to do so, which might be causing the 500 error.
